composer/utils/checkpoint.py

```diff
@@ -743,7 +768,73 @@ def save_checkpoint(
     filename: str = 'ep{epoch}-ba{batch}-rank{rank}',
     *,
     weights_only: bool = False,
+    save_planner: Optional[Any] = None,
 ) -> Union[str, None]:  # noqa: D103
+    __doc__ = f"""Checkpoint the training ``state``.
+
+    Args:
+        state (State): The training state.
+        logger (Logger): The logger.
+        filename (str): A format string describing how to name checkpoints.
+            (default: ``'ep{{epoch}}-ba{{batch}}-rank{{rank}}'``)
+
+            The following format variables are available:
```

Comment:
This won't run until the function is called, which iirc will affect the doc build. I think it has to be at root level so it is run when importing the file so it can be accessed by docs build?

Reply:
This change follows how it's done everywhere else in our codebase. Not sure which version is correct, but did this for consistency.
e.g.  https://github.com/mosaicml/composer/blob/f5986ea7604b494695d5ba632150374c3fc503d1/composer/callbacks/checkpoint_saver.py#L34

Reply:
https://mosaicml-docsite--2573.com.readthedocs.build/projects/composer/en/2573/api_reference/generated/composer.utils.save_checkpoint.html

it does not seem to work. I think fine to cleanup if you want to identify why but imo not worth it now

<img width="908" alt="image" src="https://github.com/mosaicml/composer/assets/17102158/28141dd5-8f15-42c2-ba66-9b667b9267f6">